### DISCUSSION TITLE
Invalid secret, github-token is not defined in the referenced workflow

### DISCUSSION THREAD

**Discussion by schalkneethling** [community/community#26749] [Actions]
<p>I have the following GitHub Action setup for reuse:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/mdn/workflows/blob/d971cc3696bf46ae32617e058393e45cec55ff85/.github/workflows/auto-merge.yml">
  <header class="source">

      <a href="https://github.com/mdn/workflows/blob/d971cc3696bf46ae32617e058393e45cec55ff85/.github/workflows/auto-merge.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/mdn/workflows/blob/d971cc3696bf46ae32617e058393e45cec55ff85/.github/workflows/auto-merge.yml" target="_blank" rel="noopener nofollow ugc">mdn/workflows/blob/d971cc3696bf46ae32617e058393e45cec55ff85/.github/workflows/auto-merge.yml</a></h4>

    <pre><code class="lang-yml">name: auto-merge

on: [workflow_call]

jobs:
  auto-merge:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: ahmadnassri/action-dependabot-auto-merge@v2
        with:
          target: minor
          github-token: ${{ secrets.GH_TOKEN }}
</code></pre>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I am reusing it here: <a href="https://github.com/mdn/mdn-minimalist/blob/4c2fd5a9fe02c844788001064d72d98dc2ef801b/.github/workflows/auto-merge.yml" class="inline-onebox" rel="noopener nofollow ugc">mdn-minimalist/auto-merge.yml at 4c2fd5a9fe02c844788001064d72d98dc2ef801b · mdn/mdn-minimalist · GitHub</a></p>
<p>When the action on <code>mdn-minimalist</code> is run, I get the following error:</p>
<pre><code class="lang-bash">Invalid secret, github-token is not defined in the referenced workflow
</code></pre>
<p>I am kinda stumped as it definitely looks to be defined in the referenced workflow. Any ideas? Is there a naming convention that I am missing? Thing is, the <code>auto-merge</code> action expects a <code>github-token</code> property so, I cannot simply name it <code>token</code>. Thank you in advance.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The reusable workflow needs to declare which secrets can be passed to it, see the <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onworkflow_callsecrets">documentation on <code>on.workflow_call.secrets</code></a>. That’s missing in your reusable workflow, you probably need something like this:</p>
<pre><code class="lang-yaml">on:
  workflow_call:
    secrets:
      GH_TOKEN:
        description: 'A token passed from the caller workflow'
        required: true
</code></pre>
<p>And also make sure to use the same name when passing the secret from the calling workflow.</p>

**Comment by schalkneethling** [comment]
<p>Oh brilliant! Thanks so much.</p>

**Comment by anilbey** [comment]
The working solution as of 2024: https://stackoverflow.com/a/69495861/1935611

**Reply by Gajhira-inc** [reply]
Thanks a lot aaagh nowhere you can find that parent caller must contain the
```

secrets:
  MY_SECRET: "${{secrets.MY_SECRET}}"
  

```
And that the value should be forced to a "string" Thanks a lot with this my Workflow_call is working up to Sept 2024
